source/common/ssl/context_impl.cc

```diff
@@ -490,15 +488,27 @@ ClientContextImpl::ClientContextImpl(Stats::Scope& scope, const ClientContextCon
                                      TimeSource& time_source)
     : ContextImpl(scope, config, time_source),
       server_name_indication_(config.serverNameIndication()),
-      allow_renegotiation_(config.allowRenegotiation()) {
+      allow_renegotiation_(config.allowRenegotiation()),
+      max_session_keys_(config.maxSessionKeys()) {
   if (!parsed_alpn_protocols_.empty()) {
     int rc = SSL_CTX_set_alpn_protos(ctx_.get(), &parsed_alpn_protocols_[0],
                                      parsed_alpn_protocols_.size());
     RELEASE_ASSERT(rc == 0, "");
   }
+
+  if (max_session_keys_ > 0) {
+    SSL_CTX_set_session_cache_mode(ctx_.get(), SSL_SESS_CACHE_CLIENT);
```

Comment:
Should we check or ASSERT errors for these BoringSSL calls?

Reply:
Not for those. `SSL_CTX_set_session_cache_mode()` cannot fail and returns previously configured mode, `SSL_CTX_sess_set_new_cb()` cannot fail and returns `void`.